compat/src/react-18-hooks.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useLayoutEffect, useEffect } from 'preact/hooks';
+
+/**
+ * Asynchronously schedule a callback
+ * @type {(cb: () => void) => void}
+ */
+/* istanbul ignore next */
+// Note the following line isn't tree-shaken by rollup cuz of rollup/rollup#2566
+const defer =
+	typeof Promise == 'function'
+		? Promise.prototype.then.bind(Promise.resolve())
+		: setTimeout;
+
+export function useDeferredValue(val) {
+	return val;
+}
+
+export function useTransition() {
+	const [isPending, setIsPending] = useState(false);
+
+	const start = cb => {
+		setIsPending(true);
+		cb();
+		defer(() => {
+			setIsPending(false);
+		});
+	};
+
+	return [isPending, start];
+}
+
+export const useInsertionEffect = useEffect;
```

Comment:
This is not right.

The point of insertion effects is that they run _before_ the DOM is updated. They’re where CSS-in-JS libraries should insert their style tags. If you run them as useEffect (after paint), then the user will see unstyled content.

Running them as a layout effect is also wrong. Because if the user reads layout in a layout effect then they’ll potentially get it before the style is inserted (eg by a sibling component). So the layout will be wrong.

This is why it’s a separate Hook that runs together with DOM mutations and before layout effects or regular effects. We wouldn’t be adding it if it was okay to run in the existing slots.

Reply:
Oh, when I took a look at React it seemed to be also a regular effect 😅 I must have completely misunderstood, will give it some thought

Reply:
No prob! See here:

https://reactjs.org/blog/2022/03/29/react-v18.html#useinsertioneffect

Reply:
And here https://github.com/reactwg/react-18/discussions/110

Reply:
Thank you so much for that ❤️